src/main/java/seedu/address/logic/commands/AddRecurringSessionCommand.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DURATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_FEE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INTERVAL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SUBJECT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.session.RecurringSession;
+import seedu.address.model.student.Name;
+
+
+/**
+ * Adds a recurring session to the address book.
+ */
+public class AddRecurringSessionCommand extends AddSessionCommand {
+    public static final String COMMAND_WORD = "add_rec_session";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a recurring session to the student. "
+            + "\nParameters: "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_DATE + "DATE "
+            + PREFIX_TIME + "TIME "
+            + PREFIX_DURATION + "DURATION "
+            + PREFIX_SUBJECT + "SUBJECT "
+            + PREFIX_FEE + "FEE "
+            + PREFIX_INTERVAL + "INTERVAL "
+            + PREFIX_END_DATE + "LAST DATE"
+            + "\nExample: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_DATE + "2020-01-01 "
+            + PREFIX_TIME + "12:00 "
+            + PREFIX_DURATION + "90 "
+            + PREFIX_SUBJECT + "Math "
+            + PREFIX_FEE + "40"
+            + PREFIX_INTERVAL + "7 "
+            + PREFIX_END_DATE + "2020-01-15";
+
+    public static final String MESSAGE_SUCCESS = "Added recurring session: %1$s";
+    private RecurringSession sessionToAdd;
+    private Name name;
+    /**
+     * Creates an AddRecurringSessionCommand to add the specified {@code RecurringSession}
+     */
+    public AddRecurringSessionCommand(RecurringSession recurringSession, Name name) {
```

Comment:
Maybe you would want to pull upstream first and merge inside, I believe that choon wei added a functionally that previously adding session that clashes. (E.g. If a session that starts on 2020-01-01 12.00 that last for 120mins, we can't create another session that is between  2020-01-01 12.00 to 2020-01-01 13.59) Although, with recurring session, this is much harder to achieve, because we need to take into consideration from the start to end date, if adding this rec_session will clash with future individual sessions as well. Not sure if you want to push the 2nd part into a separate issue as it seems like quite a big implementation.

Reply:
Yeah ive pulled upstream. ChoonWei's method doesn't check for Recurring session so there arent clashes.
Yep will do as a seperate issue